### PR TITLE
Fix the header scrolling by using min-height instead of height

### LINKED ISSUE #2150: The header is not fixed when scrolling to bottom
## Description
The header is not fixed. If you scroll down, you will see it disappear.

## Reproduction

 On staging.openverse.org, search for images and scroll down. You will see that after some point, the header scrolls up and disappears from the screen:

## Screenshots
Kapture 2023-02-07 at 11 23 39

## Additional context
~~Thankfully, there's an easy fix: adding `overflow-y-scroll` to `app`~~ It's not enough.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #2150 by @obulat

## Description

This PR replaces the `h-100dvh` with `min-h-100dvh` for the `app` layout to fix the header scrolling.

## Testing Instructions

Checkout the PR and test that the height is correct on all pages, both on mobile and desktop, especially iOS. 

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/layouts/default.vue [hunk 1]
```diff
@@ -208,7 +208,7 @@ export default {
 
 <style scoped>
 .app {
-  @apply h-screen h-[100dvh];
+  @apply min-h-screen min-h-[100dvh];
```
Comment (dhruvkb) [inline src/layouts/default.vue:211]: I'm still unsure how the priority of these two is set. They have the same specificity and target the same CSS property.

**Review by dhruvkb — APPROVED** [review]
This fixes the issue. 

---

But it pushes the WordPress logo below the fold, it used to fit on the screen earlier.

**Before:**



**Now:**

https://user-images.githubusercontent.com/16580576/217275664-66c041f1-fcd2-4273-a3b0-294e95ffe077.mp4

File: src/layouts/default.vue (hunk 1 above)
Comment (obulat) [reply src/layouts/default.vue:211]: If the device supports `dvh`, then the last property is used.
If it does not support `dvh`, then the last property is discarded, and h-screen (100vh) is used.

Comment (dhruvkb) [reply src/layouts/default.vue:211]: But what if a device supports `dvh` but somehow the `min-h-screen` appears later and the device prioritises that?

**Review by zackkrida — CHANGES_REQUESTED** [review]
We'll need to find a way to set `h-screen h-[100dvh]` on the full-height, yellow background pages (home, 404) and use `min-h-screen min-h-[100dvh]` on all the other routes, in order to fix this problem everywhere.

**Comment by obulat** [comment]
> We'll need to find a way to set `h-screen h-[100dvh]` on the full-height, yellow background pages (home, 404) and use `min-h-screen min-h-[100dvh]` on all the other routes, in order to fix this problem everywhere.

I guess the best way is to split the layouts again :) Here's the PR: https://github.com/WordPress/openverse-frontend/pull/2155 I think 🤞 it finally works

**Comment by zackkrida** [comment]
Closing in favor of #2155

**Pull request closed without merging**